app/src/main/res/layout/end_time.xml

```diff
@@ -0,0 +1,88 @@
+        <LinearLayout
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:padding="@dimen/spacing_extra_small"
+            android:orientation="horizontal"
+            android:visibility="@{ pickerEnd.checked ? View.VISIBLE : View.GONE }">
+
+            <TextView
+                android:layout_width="0dp"
+                android:layout_height="wrap_content"
+                android:layout_weight="2"
+                android:text="@string/sale_end_time" />
+
+            <Button
+                style="?attr/borderlessButtonStyle"
+                android:layout_width="0dp"
+                android:layout_height="wrap_content"
+                android:layout_weight="2"
+                android:textColor="@color/purple_500"
+                app:date="@{ event.endsAt }" />
+
+            <Button
+                style="?attr/borderlessButtonStyle"
+                android:layout_width="0dp"
+                android:layout_height="wrap_content"
+                android:layout_weight="1"
+                android:textColor="@color/red_500"
+                app:time="@{ event.endsAt }" />
+
+        </LinearLayout>
+
+        <LinearLayout
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:gravity="center_vertical"
+            android:orientation="horizontal"
+            android:visibility="@{ pickerEnd.checked ? View.GONE : View.VISIBLE }">
+
+            <ImageView
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:padding="@dimen/spacing_extra_small"
+                android:contentDescription="@string/date_and_time"
+                app:srcCompat="@drawable/ic_events" />
+
+            <android.support.design.widget.TextInputLayout
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:padding="@dimen/spacing_extra_small">
+
+                <android.support.design.widget.TextInputEditText
+                    android:id="@+id/end_time"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:hint="@string/sale_end_time"
+                    android:inputType="datetime"
+                    android:text="@={ event.endsAt }"
+                    app:validateEmpty="@{true}" />
+
+            </android.support.design.widget.TextInputLayout>
+        </LinearLayout>
+
+        <CheckBox
+            android:id="@+id/picker_end"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:gravity="center"
+            android:text="@string/use_date_time_picker"
+            android:textSize="@dimen/text_size_extra_small"
+            android:checked="true"/>
+    </LinearLayout>
+
+</layout>
```

Comment:
We don't want a layout for end time. How can we reuse this?

We want a general layout for the date time picker

Reply:
One has `app:date="@{ event.startsAt }"` and other `app:date="@{ event.startsAt }"`, and similar respective properties for start and end. Or do I need include both in one layout?

Reply:
You just send a different date reference in each layout. Look how you can pass data in each layout